confmap/resolver.go

```diff
@@ -65,10 +80,42 @@ func NewResolver(set ResolverSettings) (*Resolver, error) {
 		return nil, errors.New("invalid map resolver config: no URIs")
 	}
 
-	if len(set.Providers) == 0 {
+	if len(set.ProviderFactories) == 0 && len(set.Providers) == 0 {
 		return nil, errors.New("invalid map resolver config: no Providers")
 	}
 
+	if set.ProviderSettings.Logger == nil {
+		set.ProviderSettings.Logger = zap.NewNop()
+	}
```

Comment:
Should we be returning an error here or allowing future panics instead?  If the user passed in a `ResolverSettings` with no logger and a `ProviderFactories` that expects a logger, is it right to expect `NewResolver` to handle the mistake?

This check is making the code much safer to execute, but will it create confusion when not logs are written and no errors happen in this situation?

Reply:
That's a good question. The approach I'd like to take here is to make this an optional setting akin to calling an OTel language API. If you want to opt-in to logs they will be printed by some providers, otherwise it is just a no-op.

Reply:
I've updated `ProviderSettings` with a comment explaining this.